src/main/java/net/glowstone/block/blocktype/BlockType.java

```diff
@@ -303,11 +305,28 @@ public final void rightClickBlock(GlowPlayer player, GlowBlock against, BlockFac
             return;
         }
 
-        // TODO: check bounding box
         if (getMaterial().isSolid()) {
-            for (Entity entity : against.getChunk().getEntities()) {
-                if (entity instanceof LivingEntity && entity.getLocation().getBlockX() == target.getX() && entity.getLocation().getBlockY() == target.getY() && entity.getLocation().getBlockZ() == target.getZ()) {
-                    return;
+            GlowChunk[] chunks = new GlowChunk[]{
```

Comment:
It would really surprise me if an entity takes 3 chunks in a single direction. The largest entity (x/z) is  the Ender dragon (16). In any case, an entity will never have a colliding box that goes further than this.